src/parameter_window.cpp

```diff
@@ -108,6 +108,17 @@ void renderParameterWindow(const char *windowName, const std::string &curSelecte
     ImGui::End();
 }
 
+std::string getFormatStringFromStep(double step, int max_digits = 10) {
+    const double epsilon = 1e-12;  // tolerance for rounding error
+    int digits = 0;
+    double scaled = step;
+    while (digits < max_digits && std::abs(scaled - std::round(scaled)) > epsilon) {
+        scaled *= 10.0;
+        ++digits;
+    }
+    return "%." + std::to_string(digits) + "f";
+}
```

Comment:
i'm not sure if building a format string at runtime and immediately parsing it again is the best solution here, maybe we could use fmtlib (is it already used in this project?)? it should allow dynamic precision easily (see the "dynamic precision example here: https://fmt.dev/11.1/syntax/#format-examples)

Reply:
setting the format string of the slider to have the same precision as the step size is kind of a hack to account for the fact that the ImGui slider doesn't have a step setting.  It's not necessary, but it looks a lot nicer than having something like an arbitrary 6 decimal points of precision on the slider when it's just going to get snapped to the step size anyway.

I think the ImGui slider call is expecting an actual format string, so it needs to be generated regardless if it's dynamic. But if fmtlib is available it could be constructed like this, if that's preferred:

fmt::format("%.{:d}f", digits);

Reply:
> I think the ImGui slider call is expecting an actual format string

i didn't realize imgui does the format string processing internally, i thought we just provide it with a formatted string. fmt doesnt seem to be used here yet, so let's leave it like it is to keep this PR small